### ISSUE TITLE
SignIn with an account and reset the password of another acount

### ISSUE THREAD

**Issue by Dali2579** [Azure-Samples/active-directory-b2c-xamarin-native#89]
I don't know if this is a valid scenario.
I used this sample active-directory-b2c-xamarin-native.
I created my tenant in AZURE B2C.
I have already two user account on it account1  and account2
I signed in with the account1 to the application.
The application displayed account1's name correctly.
I clicked on reset password.
I entered account2's email.
I got the verification code and I updated account2's password.
when comeback to homescreen, I'm getting account2's name.
What happen to account1's session? is it logged out?
I'm not sure if this is a correct behavior.

**Comment by jennyf19** [comment]
@Dali2579 What idp are you using? B2C local accounts, FB, Twitter, etc..? are the users connected at all? Example, is it the same user w/two accounts?

Are you using the sample as-is? If so, I will try out what you explained above. Thanks.

**Comment by Dali2579** [comment]
I'm using IDP local and using the example as it is.

**Comment by jennyf19** [comment]
@Dali2579 thanks...i'll take a look

**Comment by jennyf19** [comment]
@Dali2579 Which IdP are you using?

**Comment by Dali2579** [comment]
@jennyf19 I'm only using B2C local account.

**Comment by TiagoBrenck** [comment]
@Dali2579 @jennyf19 I believe this is not an issue. The reset password flow is: type an email, then access that email (it proves that you have control of that email), then we will reset the PW for you. If you type my email, you wont be able to reset my pw, because you don't have control over my email. This reset password flow is actually found in many places and apps out there, and if you type someone else email, they will indeed get an email with a link or code, but since they didn't ask for a pw reset, they would probably ignore it.

IMHO, I don't think this is an issue.

**Comment by Dali2579** [comment]
@TiagoBrenck  Yes I agree with your analysis. my question was not about the reset password workflow. My question was in case of connecting as account1 and from this session reseting the password of account2, after the end of the reset password we need to keep connected to account1 .
below steps with screenshots
I signed in with the account1 to the application.
![image](https://user-images.githubusercontent.com/27347623/59641712-6939f180-915a-11e9-953c-3560669540e3.png)
The application displayed account1's name correctly.
![image](https://user-images.githubusercontent.com/27347623/59641809-ab633300-915a-11e9-98a9-4a56ff42b2a8.png)
I clicked on reset password.
I entered account2's email.
![image](https://user-images.githubusercontent.com/27347623/59641929-0dbc3380-915b-11e9-8a21-00a1e275ec3d.png)
I got the verification code and I updated account2's password.
![image](https://user-images.githubusercontent.com/27347623/59641988-40fec280-915b-11e9-8f63-2cefc8b7d1f9.png)
update account2  password
![image](https://user-images.githubusercontent.com/27347623/59642050-76a3ab80-915b-11e9-9a54-a8b9c1e484de.png)
when comeback to homescreen, I'm getting account2's name.
![image](https://user-images.githubusercontent.com/27347623/59642087-9a66f180-915b-11e9-9185-2f70b6d6087c.png)

as you can see account2's name is dispallyed. what happen to account1 session? is it logged out?

Thanks,
Dali

**Comment by TiagoBrenck** [comment]
Oh, now I see what you mean and indeed is an odd behavior that needs to be investigated.
Sorry about the misunderstanding.

**Comment by jennyf19** [comment]
@TiagoBrenck did you investigate this?
